src/server/game/Server/WorldSocket.cpp

```diff
@@ -358,6 +358,8 @@ WorldSocket::ReadDataHandlerResult WorldSocket::ReadDataHandler()
         }
         case CMSG_KEEP_ALIVE:
             LogOpcodeText(opcode, sessionGuard);
+            if (_worldSession)
```

Comment:
if a race condition can happen here on _worldSession then its value could become NULL after the NULL-check, causing a NULL dereference crash.
if that's the case then it's just a matter of calling sessionGuard.lock() and then sessionGuard.unlock() (or whatever is the syntax of this lock) in the CMSG_KEEP_ALIVE case ?